pipeline.lisp

```diff
@@ -15,28 +15,64 @@
    (%names :initarg :names :initform (make-hash-table :test 'equal) :accessor names))
   (:documentation "Base pipeline object that is necessary to orchestrate piping systems."))
 
+(define-condition unknown-name-error ()
+  ((%name :initarg :name :reader name)
+   (%pipeline :initarg :pipeline :reader pipeline))
+  (:documentation "Signalled if a name which was not previously set in the pipeline is looked up")
+  (:report (lambda (c stream)
+             (format stream "Name ~A is not known in pipeline ~A" (name c) (pipeline c)))))
+
+(defgeneric resolve-place (pipeline place &key ignore-unknown)
```

Comment:
It is more standard to have the parameter be `if-does-not-exist` and the permissible values be: `NIL` (return NIL) and `:error:` (return error)